WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -275,5 +281,22 @@ async Task ReadyToSignTask(AliceClient aliceClient)
 				.Take(MaxInputsRegistrableByWallet)
 				.ToShuffled()
 				.ToImmutableList();
+
+		private ImmutableList<DateTimeOffset> CreateSchedule(DateTimeOffset startTime, TimeSpan timeFrame, int numberOfEvents) =>
+			Enumerable
+				.Range(0, numberOfEvents)
+				.Select(_ => startTime.Add(0.8 * Random.NextDouble() * timeFrame))
+				.OrderBy(t => t)
+				.ToImmutableList();
+
+		private ImmutableList<Task> CreateScheduledDelays(DateTimeOffset startTime, TimeSpan timeFrame, int numberOfEvents, CancellationToken cancellationToken) =>
+			CreateSchedule(startTime, timeFrame, numberOfEvents).Select(t => CreateDelayTask(t, cancellationToken)).ToImmutableList();
+
+		private static Task CreateDelayTask(DateTimeOffset when, CancellationToken cancellationToken)
+		{
+			var timeToWait = when - DateTimeOffset.UtcNow;
+			var fixedTimeToWait = timeToWait < TimeSpan.Zero ? TimeSpan.Zero : timeToWait;
+			return Task.Delay(fixedTimeToWait, cancellationToken);
```

Comment:
ConfAwait false

Reply:
The method is not marked with `async`. So this is OK.

Reply:
Right!